esmvaltool/diag_scripts/weighting/climwip.py

```diff
@@ -418,19 +439,54 @@ def visualize_and_save_weights(weights: 'xr.DataArray', cfg: dict,
     log_provenance(caption, filename_data, cfg, ancestors)
 
 
+def parse_contributions(metric: str, cfg: dict) -> list:
+    """Return a dict of variable groups with contributions > 0."""
+    if cfg.get(f'{metric}_contributions') is not None:
+        return {
+            key: value
+            for key, value in cfg[f'{metric}_contributions'].items()
+            if value > 0
+        }
+    return {}
+
+
 def main(cfg):
     """Perform climwip weighting method."""
     models, observations = read_metadata(cfg)
 
-    variable_groups = models.keys()
+    independence_contributions = parse_contributions('independence', cfg)
+    performance_contributions = parse_contributions('performance', cfg)
+
+    if not (independence_contributions or performance_contributions):
+        errmsg = ' '.join([
+            'Either the independence_contributions or the',
+            'performance_contributions field need to be set and contain at',
+            'least one variable group with weight > 0 otherwise no weights',
+            'can be calculated!'
+        ])
+        raise IOError(errmsg)
 
     model_ancestors = []
     obs_ancestors = []
 
     performances = {}
     independences = {}
 
-    for variable_group in variable_groups:
+    for variable_group in independence_contributions.keys():
```

Comment:
I think you don't need the `.keys`
```suggestion
    for variable_group in independence_contributions:
```